### ISSUE TITLE
Create a C++ subworker library 

### ISSUE THREAD

**Issue by gavento** [substantic/rain#35]
Implement a library to make C++ subworker implementation easier. In spirit, this should consist of I/O and Context types, way to declare and register task functions and a simple mainloop function, similarly to #34. Would you sketch the API, @spirali?

**Comment by spirali** [comment]
The is my first skech of API:
```c++

#include <rainsw.h>
use namespace rainsw;

// InstanceList = LabeledList<DataInstance>

// This function takes a blob and creates a new one same content + '!' at the end.
Result<InstanceList> example_fn(Context &ctx, const InstanceList &inputs)
{
        if (inputs.size() != 1) {
		return Result::error("Function expects 1 input object");
	}
	DataObject &obj = inputs.get(0);
        if (!obj.is_blob()) {
		return Result::error("1st argument should be blob");
	}

	Vector<byte> data;
	data.assign(obj.get_ptr(), obj.size());
	data.push_back('!');

	InstanceList result;
	result.push_back(DataObject(data));
	return Result<InstanceList>(std::move(result));
}

int main() {
   Subworker subworker;
   subworker.add_fn("example", example_fn);
   subworker.run();
}
```

**Comment by spirali** [comment]
Basic C++ subworker library implemented in 437dc2b133edd412781e33ca4120666362d3813e
